docs/api/service-workers.md

```diff
@@ -0,0 +1,62 @@
+## Class: ServiceWorkers
+
+> Query and receive events from a sessions active service workers.
+
+Process: [Main](../glossary.md#main-process)
+
+Instances of the `ServiceWorkers` class are accessed by using `serviceWorkers` property of
+a `Session`.
+
+For example:
+
+```javascript
+const { session } = require('electron')
+
+// Get all service workers.
+console.log(session.defaultSession.serviceWorkers.getAllRunning())
+
+// Handle logs and get service worker info
+session.defaultSession.serviceWorkers.on('console-message', (event, messageDetails) => {
+  console.log(
+    'Got service worker message',
+    messageDetails,
+    'from',
+    session.defaultSession.serviceWorkers.getFromVersionID(messageDetails.versionId)
+  )
+})
+```
+
+### Instance Events
+
+The following events are available on instances of `ServiceWorkers`:
+
+#### Event: 'console-message'
+
+Returns:
+
+* `event` Event
+* `messageDetails` Object - Information about the console message
+  * `message` String - The actual console message
+  * `versionId` Number - The version ID of the service worker that sent the log message
+  * `source` String - The type of source for this message.  Can be `javascript`, `xml`, `network`, `console-api`, `storage`, `app-cache`, `rendering`, `security`, `deprecation`, `worker`, `violation`, `intervention`, `recommendation` or `other`.
+  * `level` Number - The log level, from 0 to 3.  In order it matches `verbose`, `info`, `warning` and `error`.
+  * `sourceUrl` String - The URL the message came from
+  * `lineNumber` Number - The line number of the source that triggered this console message
+
+Emitted when a service worker logs something to the console.
+
+### Instance Methods
+
+The following methods are available on instances of `ServiceWorkers`:
+
+#### `serviceWorkers.getAllRunning()`

```

Comment:
Would it be reasonable to add a `getAllRegistered` method here? Not necessarily in this PR, but as a followup?

Reply:
If there's a Chromium API for it, yeah, can do 👍